cookbooks/bach_krb5/recipes/upload_keytabs.rb

```diff
@@ -0,0 +1,22 @@
+require 'base64'
+# Upload keytabs to chef-server
+get_all_nodes().each do |h|
+  node[:bcpc][:hadoop][:kerberos][:data].each do |srvc, srvdat|
+    # Set host based on configuration 
+    config_host=srvdat['princhost'] == "_HOST" ? float_host(h[:hostname]) : srvdat['princhost'].split('.')[0]
+    keytab_host=srvdat['princhost'] == "_HOST" ? float_host(h[:fqdn]) : srvdat['princhost']
+
+    # Delete existing configuration item (if requested)
+    delete_config("#{config_host}-#{srvdat['principal']}") if node[:bcpc][:hadoop][:kerberos][:keytab][:recreate] == true
```

Comment:
This is from the `bcpc` cookbook but `bach_krb5/metadata.rb` does not depend on that cookbook; one will get:
`No resource or method named `delete_config' for `Chef::Recipe "upload_keytabs"'`